pydantic_extra_types/s3.py

```diff
@@ -0,0 +1,45 @@
+"""
+The `pydantic_extra_types.s3` module provides the
+[`S3Path`][pydantic_extra_types.s3.S3Path] data type.
+
+A simpleAWS S3 URLs parser.
+It also provides the `Bucket`, `Key` component.
+"""
+
+from __future__ import annotations
+
+import re
+from typing import Any, ClassVar, Type
+
+from pydantic import GetCoreSchemaHandler
+from pydantic_core import core_schema
+
+
+class S3Path(str):
+    patt: ClassVar[str] = r'^s3://([^/]+)/(.*?([^/]+)/?)$'
```

Comment:
We should have here some docstring following this pattern for example https://github.com/pydantic/pydantic-extra-types/blob/12f3f6184bad65fcc5740fe71e7a8c55f67ee37e/pydantic_extra_types/pendulum_dt.py#L33-L49

Reply:
Of course! Thanks 🫣